content/blog/posts/vectors-matrices-matrix-order.md

```diff
@@ -0,0 +1,732 @@
+
+## Matrices & vectors
+
+An \\(n \cdot m\\) matrix consists of \\(n\\) rows and \\(m\\) columns. The product of two matrices \\(\mathbf{A}\\) and \\(\mathbf{B}\\) is well defined when the number of columns in \\(\mathbf{A}\\) matches the number of rows in \\(\mathbf{B}\\). Their product is a matrix with the number of rows of \\(\mathbf{A}\\) and the number of columns of \\(\mathbf{B}\\). Thus, for a given \\(2×3\\) matrix \\(\mathbf{A}\\) to be multiplied with a given \\(3×2\\) matrix \\(\mathbf{B}\\), the product is calculated as follows to produce the resulting \\(2×2\\) matrix:
+
+$$
+\begin{array}{l@{}l}
+\mathbf{A} \cdot \mathbf{B} 
+    &{}= \begin{bmatrix} a_{11} & a_{12} & a_{13} \\\ a_{21} & a_{22} & a_{23} \end{bmatrix} \cdot \begin{bmatrix} b_{11} & b_{12} \\\ b_{13} & b_{21} \\\ b_{22} & b_{23} \end{bmatrix} \\\ \\\
+    &{}= \begin{bmatrix} a_{11}b_{11} + a_{12}b_{21} + a_{13}b_{31} & a_{11}b_{12} + a_{12}b_{22} + a_{13}b_{32} \\\ 
+a_{21}b_{11} + a_{22}b_{21} + a_{23}b_{31} & a_{21}b_{12} + a_{22}b_{22} + a_{23}b_{32} \end{bmatrix}
+\end{array}
+$$
+
+Vectors can be written as either rows:
+
+$$\vec{r} = \begin{bmatrix} r_{1} & r_{2} & r_{3} & ... & r_{m}  \end{bmatrix}$$
+
+Or columns:
+
+$$\vec{c} = \begin{bmatrix} c_{1} \\\ c_{2} \\\ c_{3} \\\ ... \\\ c_{m}  \end{bmatrix}$$
+
+The vector-matrix product follows the same rules as the matrix product, albeit treating row vectors as \\(1×m\\) matrices and column vectors as \\(m×1\\) matrices. Thus, for a given \\(2×2\\) matrix \\(\mathbf{M}\\) to be multiplied with a given \\(1×2\\) row vector \\(\vec{r}\\), we must ensure that the number of columns of the matrix on the left hand side matches the number of rows of the matrix on the right hand side. We achieve this by pre-multiplying the vector \\(\vec{r}\\) with \\(\mathbf{M}\\), resulting in a \\(1×2\\) matrix:
+
+$$
+\begin{array}{l@{}l}
+\vec{r} \cdot \mathbf{M} 
+    &{}= \begin{bmatrix} r_{1} & r_{2} \end{bmatrix} \cdot \begin{bmatrix} m_{11} & m_{12} \\\ m_{21} & m_{22}  \end{bmatrix} \\\ \\\
+    &{}= \begin{bmatrix} 
+r_{1}m_{11} + r_{2}m_{21} & 
+r_{1}m_{12} + r_{2}m_{22} \end{bmatrix}
+\end{array}
+$$
+
+In order to multiply a column vector by the same matrix, we must ensure that the number of columns of the matrix on the left hand side matches the number of rows of the matrix on the right hand side. We achieve this by post-multiplying the vector \\(\vec{c}\\) with \\(\mathbf{M}\\), resulting in a \\(2×1\\) matrix:
+
+$$
+\begin{array}{l@{}l}
+\mathbf{M} \cdot \vec{c} 
+    &{}= \begin{bmatrix} m_{11} & m_{12} \\\ m_{21} & m_{22}  \end{bmatrix} \cdot \begin{bmatrix} c_{1} \\\ c_{2} \end{bmatrix} \\\ \\\
+&{}= \begin{bmatrix} 
+m_{11}c_{1} + m_{21}c_{1} \\\ 
+m_{12}c_{2} + m_{22}c_{2} \end{bmatrix}
+\end{array}
+$$
+
+However, due to the non-commutativity of matrix multiplication, $\vec{r} \cdot \mathbf{M}$ may not equal to $\mathbf{M} \cdot \vec{c}$ as the two products are necessarily not equivalent:
```

Comment:
```suggestion
Since \\(\vec{c}\\) is the transpose of \\(\vec{r}\\), and transpose multiplication follows the identity \\((\mathbf{A}\mathbf{B})^\mathbf{T} = \mathbf{B^T}\mathbf{A^T}\\), we need to transpose \\(\mathbf{M}\\) and reverse the order in the multiplication to get the equivalent product as a column vector. If we didn't reverse the order and take the transpose, we would get a different result:
```